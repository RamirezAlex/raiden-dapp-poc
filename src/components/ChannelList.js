import React from 'react'
import ChannelOpenItem from './ChannelOpenItem'

const ChannelList = (props) => {
  return (
    <table>
      <tbody>
        <tr>
          <th>#</th>
          <th>Channel Identifier:</th>
          <th>Participant1:</th>
          <th>Participant2:</th>
        </tr>
        {
          props.channelsOpened.map((channel, key) => (
            <ChannelOpenItem channel={channel} count={key} key={key} />
          ))
        }
      </tbody>
    </table>
  )
}

export default ChannelList
