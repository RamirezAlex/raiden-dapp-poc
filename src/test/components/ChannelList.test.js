import React from 'react'
import ChannelList from '../../components/ChannelList'
import { create } from 'react-test-renderer'

describe('ChannelList', () => {
  it('Shoulde render a EventList', () => {
    expect(ChannelList).not.toBeNull()
    const channelsOpened = [
      {
        channelIdentifier: 24,
        participant1: '0x0E809A051034723beE67871a5A4968aE22d36C5A',
        participant2: '0x504300C525CbE91Adb3FE0944Fe1f56f5162C75C'
      },
      {
        channelIdentifier: 27,
        participant1: '0x2046F7341f15D0211ca1EBeFb19d029c4Bc4c4e7',
        participant2: '0x32bBc8ba52FB6F61C24809FdeDA1baa5E55e55EA'
      },
      {
        channelIdentifier: 36,
        participant1: '0xAa445ec9d7c98D850e01d45423B75ADf5de0b256',
        participant2: '0x32bBc8ba52FB6F61C24809FdeDA1baa5E55e55EA'
      }
    ]
    const component = create(<ChannelList channelsOpened={channelsOpened} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
