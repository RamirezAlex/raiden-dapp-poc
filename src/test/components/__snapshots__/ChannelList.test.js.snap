// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelList Shoulde render a EventList 1`] = `
<table>
  <tbody>
    <tr>
      <th>
        #
      </th>
      <th>
        Channel Identifier:
      </th>
      <th>
        Participant1:
      </th>
      <th>
        Participant2:
      </th>
    </tr>
    <tr>
      <td>
        0
      </td>
      <td>
        24
      </td>
      <td>
        0x0E809A051034723beE67871a5A4968aE22d36C5A
      </td>
      <td>
        0x504300C525CbE91Adb3FE0944Fe1f56f5162C75C
      </td>
    </tr>
    <tr>
      <td>
        1
      </td>
      <td>
        27
      </td>
      <td>
        0x2046F7341f15D0211ca1EBeFb19d029c4Bc4c4e7
      </td>
      <td>
        0x32bBc8ba52FB6F61C24809FdeDA1baa5E55e55EA
      </td>
    </tr>
    <tr>
      <td>
        2
      </td>
      <td>
        36
      </td>
      <td>
        0xAa445ec9d7c98D850e01d45423B75ADf5de0b256
      </td>
      <td>
        0x32bBc8ba52FB6F61C24809FdeDA1baa5E55e55EA
      </td>
    </tr>
  </tbody>
</table>
`;
